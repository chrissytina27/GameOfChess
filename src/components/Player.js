import React from "react";
import PropTypes from "prop-types";
import GraveyardPiece from "./GraveyardPiece";

const Player = props => {
  const renderGraveyard = graveyard => {
    if (!graveyard || !graveyard.length) return;
    return graveyard.map((piece, index) => {
      return (
        <div key={index} className="graveyard">
          <GraveyardPiece
            piece={piece}
            enabled={props.graveyardPick}
            resurrectPiece={props.resurrectPiece}
          />
        </div>
      );
    });
  };

  return <div>{renderGraveyard(props.graveyard)}</div>;
};
Player.propTypes = {
  name: PropTypes.string,
  graveyard: PropTypes.array,
  graveyardPick: PropTypes.bool,
  resurrectPiece: PropTypes.func
};
export default Player;
