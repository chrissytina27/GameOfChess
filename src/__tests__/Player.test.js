import Player from "../components/Player";
import React from "react";
import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer";

describe("Player", () => {
  const testProps = {
    name: "christina",
    graveyard: ["a", "b", "c"],
    graveyardPick: false,
    resurrectPiece: jest.fn(() => ({}))
  };

  it("renders correctly", () => {
    const tree = renderer.create(<Player {...testProps} />).toJSON();
    expect(tree.children.length).toBe(testProps.graveyard.length);
    expect(tree).toMatchSnapshot();
  });
});
