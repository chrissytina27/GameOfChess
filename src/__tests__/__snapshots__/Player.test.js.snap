// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Player renders correctly 1`] = `
<div>
  <div
    className="graveyard"
  >
    <div
      className="graveyard-piece"
      onClick={[Function]}
    />
  </div>
  <div
    className="graveyard"
  >
    <div
      className="graveyard-piece"
      onClick={[Function]}
    />
  </div>
  <div
    className="graveyard"
  >
    <div
      className="graveyard-piece"
      onClick={[Function]}
    />
  </div>
</div>
`;
